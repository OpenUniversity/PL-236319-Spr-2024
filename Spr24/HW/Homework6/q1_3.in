use_module(library(yall)).
set_prolog_flag(answer_write_options, [quoted(true), portray(true), max_depth(0), spacing(next_argument)]).
print("===TEST START===").

{L}/(
    findall(
        [X],
        add_bt(bt(5, []), [bt(4, [])], X),
        L1
    ),
    msort(L1, L)
).

{L}/(
    findall(
        [X],
        add_bt(bt(3, [bt(2, [])]), [empty, bt(4, [bt(5, [])])], X),
        L1
    ),
    msort(L1, L)
).

{L}/(
    findall(
        [X],
        add_bt(bt(1, []), [empty, empty, bt(3,[bt(4,[bt(5,[])]), bt(2,[])])], X),
        L1
    ),
    msort(L1, L)
).