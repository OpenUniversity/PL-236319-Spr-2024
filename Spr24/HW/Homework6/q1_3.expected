true.

L = [[[empty, bt(4, [bt(5, [])])]]].

L = [[[empty, empty, bt(3, [bt(4, [bt(5, [])]), bt(2, [])])]]].


L = [[[bt(1, []), empty, bt(3, [bt(4, [bt(5, [])]), bt(2, [])])]]].


